packages/suite/src/views/wallet/send/components/Outputs/index.tsx

```diff
@@ -1,108 +1,79 @@
-    display: flex;
-    flex-direction: column;
-    margin: 32px 42px;
-    margin-bottom: 20px;
-
-    &:last-child {
-        margin-bottom: 0;
-    }
-
-    @media (max-width: ${variables.SCREEN_SIZE.SM}) {
-        margin: 32px 20px;
-    }
-
-    ${props =>
-        props.index > 0 &&
-        css`
-            margin: 0 42px;
-            padding-top: 32px;
-            border-top: 1px solid ${({ theme }) => theme.STROKE_GREY};
-        `}
-`;
-
-const Row = styled.div`
-    padding: 0 0 10px;
-
-    &:last-child {
-        padding: 0;
+    > div {
+        display: flex;
+        flex-direction: column;
+        gap: ${spacingsPx.md};
     }
 `;
 
 interface OutputsProps {
     disableAnim?: boolean; // used in tests, with animations enabled react-testing-library can't find output fields
 }
 
-const Outputs = ({ disableAnim }: OutputsProps) => {
-    const { outputs } = useSendFormContext();
-    const [renderedOutputs, setRenderedOutputs] = useState(1);
-    const lastOutputRef = useRef<HTMLDivElement | null>(null);
-
-    const onAddAnimationComplete = () => {
-        // scrolls only on adding outputs, doesn't scroll on removing them
-        if (outputs.length > 1 && outputs.length > renderedOutputs) {
-            lastOutputRef?.current?.scrollIntoView({ behavior: 'smooth' });
-        }
-
-        setRenderedOutputs(outputs.length);
-    };
+export const Outputs = ({ disableAnim }: OutputsProps) => {
+    const [height, setHeight] = useState(0);
 
-    useEffect(() => {
-        if (outputs.length < renderedOutputs) {
-            // updates rendered outputs count when removing an output
-            // this is necessary because onAddAnimationComplete is not fired when removing 2nd output
-            setRenderedOutputs(outputs.length);
-        }
-    }, [outputs.length, renderedOutputs, setRenderedOutputs]);
+    const { outputs } = useSendFormContext();
+    const renderCount = useRendersCount();
+    const ref = useRef<HTMLDivElement>(null);
 
-    const animation = outputs.length > 1 && !disableAnim ? motionAnimation.expand : {}; // do not animate if there is only 1 output, prevents animation on clear
+    useLayoutEffect(() => {
+        setHeight(ref.current?.offsetHeight || 0);
+    }, [outputs]);
 
     return (
-        <AnimatePresence initial={false}>
-            <Wrapper>
+        <Container $height={height || 0}>
+            <div ref={ref}>
                 {outputs.map((output, index) => (
                     <motion.div
+                        layout
                         key={output.id}
-                        {...animation}
-                        onAnimationComplete={onAddAnimationComplete}
+                        initial={
+                            // 3 seems to be the initial amount of renders for this component,
+                            // no entrance animation on the first render
+                            index === 0 || renderCount < 4 || disableAnim
```

Comment:
I don't know but counting renders seems a bit hacky to me. Can't we do it somehow better?

Reply:
I am not sure and I've sent a dumb amount of time on making this animation work >_< Form id logic was interfering with `AnimatePresence` and i did not want to touch the form hooks much. Also I cannot just use some ref and turn off the animation on the first render because it does not happen on the first render. If you have ideas – shoot! 🙌

Reply:
I would expect that the library could offer some handler for it. Don't worry about it, if it works, it's OK. Maybe just please mark it as TODO for the future

Reply:
I believe I found the cause and changed the approach! fb59d7f67a

It's probably [this](https://github.com/trezor/trezor-suite/blob/f4f7bf0dea3c04bf1bd4362b6e6cefcca6304f18/packages/suite/src/hooks/wallet/useSendForm.ts#L303) effect. The default state should be loaded synchronously, not as a hook with a reset. That logic also causes a momentary blink...